mono/metadata/domain.c

```diff
@@ -725,6 +725,10 @@ mono_init_internal (const char *filename, const char *exe_filename, const char *
                 mono_defaults.corlib, "System.Threading", "ThreadAbortException");
 	g_assert (mono_defaults.threadabortexception_class != 0);
 
+	mono_defaults.appdomainunloadedexception_klass = mono_class_from_name (
+	            mono_defaults.corlib, "System", "AppDomainUnloadedException");
+	g_assert (mono_defaults.appdomainunloadedexception_klass);
```

Comment:
The code is asserting that the class exists though, so the runtime will blow up nonetheless.